app/scripts/models/auth_brokers/fx-desktop-v2.js

```diff
@@ -64,15 +46,7 @@ define(function (require, exports, module) {
     fetch () {
       return proto.fetch.call(this).then(() => {
         if (! this.environment.isAboutAccounts()) {
-          // The default behavior of FxDesktop brokers is to halt before
-          // the signup confirmation poll because about:accounts takes care
-          // of polling and updating the UI. However if we are not in about:accounts
-          // we do not want the halting behavior.
-          this._behaviors.keys().forEach((behaviorName) => {
```

Comment:
The hard coding of which behavior to use always made me slightly uncomfortable, we should have been returning the `defaultBehavior` for each of the behavior instead. The whole scheme was a bit crap, so I used a capability and behavior combination, which worked out to be cleaner.